### PR TITLE
Allow triple left click on a  TextBox to select all text

### PR DESCRIPTION
**Pull request by JLdgu**
Triple clicking in a textbox to select all text is a common action on Windows applications

### CODE REVIEW

File: src/MaterialDesignThemes.Wpf/TextFieldAssist.cs [hunk 1]
```diff
@@ -150,6 +150,45 @@ public static class TextFieldAssist
...
+        (bool)obj.GetValue(SelectAllOnTripleClickProperty);
+
+    public static void SetSelectAllOnTripleClick(DependencyObject obj, bool value) =>
+        obj.SetValue(SelectAllOnTripleClickProperty, value);
+
+    private static void OnSelectAllOnTripleClickChanged(
+        DependencyObject d,
+        DependencyPropertyChangedEventArgs e)
+    {
+        if (d is TextBox textBox)
```
Comment (corvinsz) [inline src/MaterialDesignThemes.Wpf/TextFieldAssist.cs]:
```suggestion
        if (d is TextBoxBase textBox)
```

File: src/MaterialDesignThemes.Wpf/TextFieldAssist.cs [hunk 2]
```diff
@@ -150,6 +150,45 @@ public static class TextFieldAssist
...
+            else
+                textBox.PreviewMouseLeftButtonDown -= TextBox_PreviewMouseLeftButtonDown;
+        }
+    }
+
+    private static void TextBox_PreviewMouseLeftButtonDown(
+        object sender,
+        MouseButtonEventArgs e)
+    {
+        if (e.ClickCount == 3 && sender is TextBox textBox)
```
Comment (corvinsz) [inline src/MaterialDesignThemes.Wpf/TextFieldAssist.cs]:
```suggestion
        if (e.ClickCount == 3 && sender is TextBoxBase textBox)
```

**Review by corvinsz — CHANGES_REQUESTED** [review]
I few things I noticed:
1. We should type check against `TextBoxBase` so this behavior also works for the `RichTextBox`
2. Maybe an example dedicated to this attached property with a small description would be helpful, especially because this behavior is not something consumers can "just see". Please don't forget the MD3 demo app 😄 
3. In my opinion the test you added should also test the negative (`materialDesign:TextFieldAssist.SelectAllOnTripleClick="False"`) behavior to prevent any regressions.

**Review by corvinsz — APPROVED** [review]
LGTM!

The tests could be made data-driven, but I’m totally fine with the current approach as well.

**Review by Keboo — APPROVED** [review]
This is great, thank you! For some reason I thought this was already the default. I tweaked it slightly so that it will be on by default for anyone using the library but let people opt out with the AP.
